xinference/model/llm/llm_family.py

```diff
@@ -782,6 +782,25 @@ def get_user_defined_llm_families():
         return UD_LLM_FAMILIES.copy()
 
 
+def match_model_size(
+    model_size: Union[int, str], spec_model_size: Union[int, str]
```

Comment:
Now what if `model_size` is a float value? For example, `model_size` is `1.8` and `spec_model_size` is `1_8`?

Reply:
OK, I will add the support. I noticed the type hint is `int` or `Union[int, str]` in many function signatures, I will modify them as well.

Reply:
I think only int and string will be passed as parameter of RESTful API endpoint for launch model after I read the source code, the float value will be converted to string by replace the dot to underscore like "1_8", so the function above just supports int and str is sufficient.

The RESTful endpoint for model launch will convert string to int if there is no underscore in it, it will make "1.8" format float string failed, so I added the support for float string with dot as delimiter.

And I found launch model APIs in `restful_api` only support int and string format value, so I added the support for float since to use float value is more intuitive.